### DISCUSSION TITLE
Expose `Geometry2D::bresenham_line()` to scripting

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#11004] [2D]
[Bresenham's line algorithm](https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm) is useful when implementing grid-based drawing tools and doing line-of-sight calculations in roguelike games.

This algorithm is already implemented as [`Geometry2D::bresenham_line()`](https://github.com/godotengine/godot/blob/44fa552343722bb048e2d7c6d3661174a95a8a3c/core/math/geometry_2d.h#L454). I think we can expose it to scripting.

**Comment by dalexeev** [comment]
* See https://github.com/godotengine/godot/pull/74714.
